### PR TITLE
created type file for Noteflight client API

### PR DESCRIPTION
**Pull request by nickslick03**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
-  [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/noteflight-client/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+
+type EventFilter<T, Event, Props> = T extends Event
+? Props
+: T extends 'any'
+? Partial<Props>
+: {};
+
+interface NoteflightEventProps {
+    /**
+     * 	A string value that identifies the kind of event that was dispatched.
```
Comment (DangerBotOSS) [inline types/noteflight-client/index.d.ts:125]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by peterblazejewicz** [comment]
Please adopt formatting enforced on the project:
https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes
run:
```bash
npm run prettier -- --write types/noteflight-client/**/*.ts
```

**Comment by jakebailey** [comment]
This isn't a package on npm, right?

**Comment by nickslick03** [comment]
No, it's from noteflight, the music notation website.

https://www.noteflight.com/info/api/client_doc_v2

File: types/noteflight-client/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,664 @@
...
+type NoteflightEvent<T> = NoteflightEventProps &
+    EventFilter<T, 'scoreDataLoaded', ScoreDataLoadedProps> &
+    EventFilter<T, 'selectionChange', SelectionChangeProps> &
+    EventFilter<T, 'playbackRequest' | 'playbackStop', PlaybackProps>;
+
+type EventFilter<T, Event, Props> = T extends Event ? Props : T extends 'any' ? Partial<Props> : {};
+
+interface NoteflightEventProps {
+    /**
+     * 	A string value that identifies the kind of event that was dispatched.
```
Comment (DangerBotOSS) [inline types/noteflight-client/index.d.ts:127]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Review by jakebailey — APPROVED** [review]

**Comment by nickslick03** [comment]
Ready to merge

**Pull request merged**